src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -95,5 +97,33 @@ public static class IOExtensions {
             }
             return Path.DirectorySeparatorChar.ToString();
         }
+
+        public static bool HasWritePermission(this string path) {
```

Comment:
OK. I wanted stricter check since writing generally assumes reading (in common sense case anyway)